app/models/grda_warehouse/census/project_batch.rb

```diff
@@ -18,29 +20,16 @@ def initialize(start_date, end_date)
     end
 
     def build_census_batch
-      GrdaWarehouse::Census.census_populations.each do |population|
-        scope = GrdaWarehouse::ServiceHistoryEnrollment.public_send(population[:population])
-        add_clients_to_census_buckets(
-          get_client_and_project_counts(scope),
-          population[:population]
-        )
+      populations = GrdaWarehouse::Census.census_populations.map { |p| p[:population] }.uniq
+      populations.each do |population|
+        scope = GrdaWarehouse::ServiceHistoryEnrollment.public_send(population)
+        add_clients_to_census_buckets(get_client_and_project_counts(scope), population)
       end
 
-      @by_count.each do | project_id, census_collection |
+      @by_count.each do |project_id, census_collection|
         inventories = GrdaWarehouse::Hud::Project.find(project_id).inventories.within_range(@start_date..@end_date)
-        census_collection.each do | date, census_item |
-          filtered_inventory = inventories.select do |inventory|
```

Comment:
refactored into inventory_active_on_date

Reply:
Thank you!